### PR TITLE
Add Helm pipeline and tasks

### PR DESCRIPTION
**Pull request by a-roberts**
Both secure and insecure ones (one requiring a Helm secret, the other not; if we can do control statements that'd be cool but pretty sure we can't with tektoncd/pipelines atm)

Limitations for these pipelines and tasks: happy to hold off on merging until we make it awesome:

- Must have replacemerepo as your repository image and replacemetag as your repository tag. 
- Chart must be under `chart` folder .
- Helm chart folder name e.g. `chart/foo` must match repository name e.g. `github.com/myorg/foo/chart/foo`.
- Uses Helm 2.9.1 for compatibility with Microclimate projects and clusters running said Tiller version such as IBM Cloud Private 3.x.

### CODE REVIEW

**Comment by ncskier** [comment]
One thing you could do to address your first two limitations `Must have replacemerepo as your repository image and replacemetag as your repository tag.` and `Chart must be under chart folder .`, is to make the `chart` folder and the substitute words (`replacemerepo` and `replacemetag`) parameters in your pipeline.

For example
```
params:
...
- name: chart-dir
  description: The relative path of the chart directory
  default: chart
- name: replace-repo
  description: The placeholder image to be replaced
  default: replacemerepo
- name: replace-tag
  description: The placeholder tag to be replaced
  default: replacemetag
```

I know this doesn't exactly "fix" the limitation, but it would alleviate some of the constraints. What do you think?

File: config/helm-deploy-task-insecure.yaml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+apiVersion: tekton.dev/v1alpha1
+kind: Task
+metadata: 
+  name: deploy-simple-helm-task-insecure # No Helm secret required
+spec:
+  inputs: 
+    resources: 
+    - name: git-source
+      type: git
+    - name: image-out
```
Comment (ncskier) [inline config/helm-deploy-task-insecure.yaml]: Maybe I missed something, but is this `image-out` used somewhere in the task? It looks like all of the image related information is passed in as parameters.

If this is the case, then it raises the question should we use an image PipelineResource or Parameters to pass around the image information?

**Review by ncskier — CHANGES_REQUESTED** [review]
Looks great @a-roberts. I left a comment about an idea to address the first two limitations you listed. Also, I have one question below about the `image-out` and passing around image information.

As you said before, it's a shame that we can't have control flows in these pipelines. I looked into [ordering](https://github.com/tektoncd/pipeline/blob/master/docs/pipelines.md#ordering), but that won't help. Also, since we mount the volume secret, we can't use bash control flow, because in the insecure case the task would fail to mount an absent secret. So, for now we have to make do with a lot of duplicate text.

One other addition that would be very nice would be adding to the README some directions on how to use this pipeline. Also a link to a sample app with a helm chart that people can run if they want to get started with using this pipeline (and I can use it to test this code 😄 ). We don't have this kind of documentation for any of the other Pipelines yet, but I think we need to start somewhere. You can decide how detailed to make the README; you've probably been using the aforementioned resources  to test on your own, so at a minimum you can just paste these into the README. Also, we will want to document the limitations that you listed.

File: config/helm-deploy-task-insecure.yaml (hunk 1 above)
Comment (a-roberts) [reply config/helm-deploy-task-insecure.yaml]: Isn't used, removing

**Comment by a-roberts** [comment]
Thanks @ncskier! I've parameterised everything and have an example Node.js app on GHE I've been using. 

I'll publish a noddy sample to my public Github and reference this in the readme:  "quick start" example, then run the pipeline against it manually, replacing values (as I won't be able to leverage the extension due to our laptops not being exposed to the outside world).

**Comment by a-roberts** [comment]
@ncskier I've created https://github.com/a-roberts/simple-node which is the tiniest Node.js app i can think of with a chart, a Dockerfile and it's a simple listen server with a liveness probe. I'm working on the docs now so please take a look at the readme to see what I'm thinking: hopefully my example all works by the time you next take a look.

**Comment by a-roberts** [comment]
My example works with both the manually created PipelineResources and PipelineRun using our Helm pipeline and task, I've provided a convenience script for secrets.

File: runner.yaml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      value: "simple-node"
+    - name: release-name
+      value: "simple-node-release"
+    - name: target-namespace
+      value: "default"
+    - name: image-name
+      value: "<mydockerusername>/my-simple-node-app"
+    - name: image-tag
+      value: "latest"
+
```
Comment (ncskier) [inline runner.yaml]: Is this blank line supposed to be here?

File: runner.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  trigger:
+    type: manual
+  serviceAccount: 'default'
```
Comment (ncskier) [inline runner.yaml]: Can this be double quotes to match the other quotes in this file?

**Review by ncskier — CHANGES_REQUESTED** [review]
@a-roberts Just a couple nits. Looks great, and got your example running on my laptop!

**Review by ncskier — APPROVED** [review]

**Pull request merged**